code-challenges.js

```diff
@@ -14,36 +14,168 @@ const fruit2 = "banana"
 
 const currentCohort = "Charlie 2022"
 
+//First declare the function (stringReverse)
+const stringReverse = (string1) => {
+    //Function will use split method to turn the string into an array. I use """ to split every character to its own array element. Since the split() method is an accessor, I have to assign it to a new variable.
+    let stringArray = string1.split("")
+    //The method reverse() is a mutator so the original array will be changed
+    stringArray.reverse()
+    //I declare a new variable (strRev) to store the output from the join() method because it is an accessor
+    let strRev = stringArray.join("")
+    return strRev
+}
+
+//Input is currentCohort which has a data type of string ==> Output: "2202 eilrahC"
+console.log(stringReverse(currentCohort))
+
 
 // --------------------4) Create a statement that will evaluate whether each number in an array is even or odd. Use the test variable provided below. Expected output: "odd" "even" "odd" "even" "odd" "even"
 
 const myArray = [13, 34, 5, 10, 27, 42]
 
+//The conditional statement we would need, would check the value of the given index for myArray and determine if it was either even or odd.
+//The first condition would check to see if the given value at the given index was even. Every other value would be odd. I have commented out the code to show a loop implementation of the function created with the conditional statement.
+
+// I know that (any number) % 2 that outputs 0, is an even number
+
+//  if (myArray[i] % 2 === 0) {
+//    console.log("even")
+//  } else {
+//     console.log("odd")
+//  }
+
+
+//Function will be used in a for loop to iterate each element of the array and use if else conditional statements to determine if odd or even. The function will then return a string "odd" or "even"
+
+//First declare the function (oddOrEven) with a single parameter (array1)
+//The function is declared outside of the for loop code block. The loop will pass the value of each index to the function. 
+//We will declare arrayAtIndex with let so that it can be scoped locally in the code block of the for loop.
+
+const oddOrEven = (arrayAtIndex) => {
+    //This conditional statement checks if the array element at the given index is even and returns the string "even"
+     if (arrayAtIndex % 2 === 0) {
+         return "even"
+     //All other conditions will yield an odd number and return the string "odd"
+     } else {
+         return "odd"
+     }
+ }
+//Start at index 0 and iterate for the length of the array 1 step at at time.
+//Input is the array stored in the constant myArray ==> Output will be odd, even, odd, even, odd, even
+ for (let i = 0; i < myArray.length; i++) {
+
+        let arrayAtIndex = myArray[i]
+        //Every iteration arrayAtIndex will be overwritten and invoke the function
+        console.log(oddOrEven(arrayAtIndex))
+     }
```

Comment:
Your pseudocode and research are great. Good work on completing this code challenge section